stats.c

```diff
@@ -1676,26 +1790,57 @@ int main_stats(int argc, char *argv[])
     bam1_t *bam_line = bam_init1();
     if ( optind<argc )
     {
-        // Collect stats in selected regions only
-        hts_idx_t *bam_idx = sam_index_load(info->sam,bam_fname);
-        if (bam_idx == 0)
-            error("Random alignment retrieval only works for indexed BAM files.\n");
-
-        int i;
-        for (i=optind; i<argc; i++)
-        {
-            hts_itr_t* iter = bam_itr_querys(bam_idx, info->sam_header, argv[i]);
-            while (sam_itr_next(info->sam, iter, bam_line) >= 0) {
-                if (info->split_tag) {
-                    curr_stats = get_curr_split_stats(bam_line, split_hash, info, targets);
-                    collect_stats(bam_line, curr_stats);
+        if(targets)
+            error("Both a target file and target regions have been specified. Please choose only one method.\n");
```

Comment:
Complex though it is, specifying both regions and targets is quite common place in some other commands.

Historically regions have been used to do index skipping with targets files doing filtering while streaming, so if there are a lot of regions to work over for chr20 the user may use `20` as the region (to stream the whole chromosome) with the targets file to filter the output.